### PR TITLE
[ADD] odoo 10 portage of base_phone_validation

### PR DESCRIPTION
**Pull request by ventilooo**
A v10 migration of #42

### CODE REVIEW

File: base_phone_validation/__init__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
```
Comment (leemannd) [inline base_phone_validation/__init__.py]: Short headers

**Review by leemannd — APPROVED** [review]
Thanks. LGTM.
Only nitpicking. Could you update your pr to short headers?

File: base_phone_validation/models/res_partner.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        :param phonenumber: string
+        :param str fieldname: name of the field the number is related to.
+        :raise ValidationError: if the given number is not validated.
+        """
+        if self.country_id:
+            number = phonenumbers.parse(phonenumber, self.country_id.code)
+            if not phonenumbers.is_valid_number_for_region(
+                    number, self.country_id.code):
+                error_msg = u'\n'.join([
+                    _(u'The number ({}) "{}" seems not valid for {}.').format(
```
Comment (rimbendhaou) [inline base_phone_validation/models/res_partner.py:33]: @ventilooo In this error message you display the country name and in the other case (elif self.env.user.company_id.country_id:), the country code is displayed. Better to have the same message in all cases

**Comment by ventilooo** [comment]
@leemannd ping ?

File: base_phone_validation/models/res_partner.py (hunk 2 above)
Comment (ventilooo) [reply base_phone_validation/models/res_partner.py:33]: fixed

**Comment by foutoucour** [comment]
Hello,

Do you think we could merge this pull request?

**Comment by ventilooo** [comment]
@alexis-via 

It seems @foutoucour need that piece of code  :smile:

**Comment by foutoucour** [comment]
Hey, it has been a week already ;(
Could anyone from OCA merge this approved PR? (@rimbendhaou @leemannd) Thank you

**Comment by leemannd** [comment]
@foutoucour There is only one positive review (mine) you could try to add your review (functional/code review). It will help to have it merged.
By the way, I have no rights for merge

**Review by rimbendhaou — APPROVED** [review]
lgtm

**Review by foutoucour — APPROVED** [review]

File: base_phone_validation/i18n/base_phone_validation.pot [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# Translation of Odoo Server.
```
Comment (pedrobaeza) [inline base_phone_validation/i18n/base_phone_validation.pot:1]: Remove this file

File: base_phone_validation/README.rst [hunk 4]
```diff
@@ -0,0 +1,8 @@
+Base Phone Validation
```
Comment (pedrobaeza) [inline base_phone_validation/README.rst:1]: As the README misses some parts, you can take the occasion for using the new fragments system. Check OCA/maintainer-tools for the template

File: base_phone_validation/models/res_partner.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+                        fieldname, phonenumber, local_country.name
+                    ),
+                    _(u'Please double check it.')
+                ])
+                raise ValidationError(error_msg)
+
+    @api.constrains('phone', 'country_id')
+    def _phone_number_validation(self):
+        for rec in self:
```
Comment (pedrobaeza) [inline base_phone_validation/models/res_partner.py:56]: For avoiding one indentation level you can put

```
...self.filtered ('phone')
```

File: base_phone_validation/models/res_partner.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+            if rec.phone:
+                rec._force_validation(rec.phone, 'phone')
+
+    @api.constrains('fax', 'country_id')
+    def _fax_number_validation(self):
+        for rec in self:
```
Comment (pedrobaeza) [inline base_phone_validation/models/res_partner.py:62]: The same about filtered

File: base_phone_validation/models/res_partner.py [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+            if rec.fax:
+                rec._force_validation(rec.fax, 'fax')
+
+    @api.constrains('mobile', 'country_id')
+    def _mobile_number_validation(self):
```
Comment (pedrobaeza) [inline base_phone_validation/models/res_partner.py:67]: The same about filtered

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Pull request closed without merging**